src/result-generation.php

```diff
@@ -2,357 +2,225 @@
+            // Sanitize and store the position from the URL
+            $position = htmlspecialchars($_GET['position_id']);
+
             // Fetch candidates for the specified position from the database
             $query = "SELECT * FROM candidate WHERE position_id = ?";
-            $stmt = mysqli_prepare($connection, $query);
-
-            // Check if the prepared statement is valid
-            if ($stmt) {
-                mysqli_stmt_bind_param($stmt, "s", $position);
-                mysqli_stmt_execute($stmt);
-                $result = mysqli_stmt_get_result($stmt);
-
-                // Check if the result set is valid
-                if ($result) {
-                    if (mysqli_num_rows($result) > 0) {  
-                        // Create an array to store candidates and their respective vote counts
-                        $candidates = array();
-
-                        // Fetch and store vote counts for each candidate
-                        while ($candidate_data = mysqli_fetch_assoc($result)) {
-                            $candidate_id = $candidate_data['candidate_id'];
-                            $vote_count_query = mysqli_query($connection, "SELECT COUNT(*) as vote_count FROM vote WHERE candidate_id = $candidate_id");
-                            $vote_result = mysqli_fetch_assoc($vote_count_query);
-                            $vote_count = $vote_result['vote_count'];
-
-                            // Store candidate information and vote count in the array
-                            $candidates[] = array(
-                                'candidate_data' => $candidate_data,
-                                'vote_count' => $vote_count
-                            );
-                        }
-
-                        // Sort candidates based on their vote count (descending order)
-                        usort($candidates, function($a, $b) {
-                            return $b['vote_count'] - $a['vote_count'];
-                        });
-
-                        // Separate the candidate with the highest vote count
-                        $highest_vote_candidate = array_shift($candidates);
-
-                        // Display candidate with the highest vote count separately
-                        $candidate_data = $highest_vote_candidate['candidate_data'];
-                        $vote_count = $highest_vote_candidate['vote_count'];
+            $stmt = $connection->prepare($query);
+            $stmt->bind_param("s", $position);
+            $stmt->execute();
+            $result = $stmt->get_result();
+
+            // Check if candidates exist for the position
+            if ($result->num_rows > 0) {
+                // Loop through candidates
+                while ($candidate_data = $result->fetch_assoc()) {
+                    $candidate_id = $candidate_data['candidate_id'];
+                    $vote_count_query = "SELECT COUNT(*) as vote_count FROM vote WHERE candidate_id = $candidate_id";
```

Comment:
Use prepared sql statements next time. This is to avoid possible SQL injections to you r code.